src/main/java/com/productService/service/ProductServiceImpl.java

```diff
@@ -16,76 +17,78 @@
 class ProductServiceImpl implements ProductService {
 
     private final ProductDao productDao;
+    private final UserDao userDao;
 
     @Autowired
-    public ProductServiceImpl(ProductDao productDao) {
+    public ProductServiceImpl(ProductDao productDao, UserDao userDao) {
         this.productDao = productDao;
+        this.userDao = userDao;
     }
 
     /**
      * This function is creating product
+     *
      * @param product
      */
     @Override
-    public void createProduct(Product product) {
+    public Optional<Product> createProduct(Product product) {
         log.info("Entering ProductServiceImpl.createProduct");
-        productDao.save(product);
+        return productDao.save(product);
     }
 
     /**
-     *  This function is fetching all products
+     * This function is fetching all products
+     *
      * @return
      */
     @Override
     public List<Product> getAllProducts() {
         log.info("Entering ProductServiceImpl.getAllProducts");
         return productDao.getAllProducts();
-
     }
 
     /**
      * This function is fetching product by Id
+     *
      * @param id
      * @param sellerDetail
      * @return
      */
     @Override
-    public Optional<Product> findProductById(String id, boolean sellerDetail) {
+    public Optional<Product> findProductById(String id, boolean sellerDetail) throws DependencyFailureException {
```

Comment:
its not throwing `DependencyFailureException` anymore

Reply:
List<User> getUserByUserName(String userName) throws DependencyFailureException method is throwing so i have to add DependencyFailureException  in the method.

Reply:
Added before method signature  Optional<Product> findProductById(String id, boolean sellerDetail) throws DependencyFailureException;